pkg/controller/rsm/transformer_object_generation.go

```diff
@@ -715,7 +715,9 @@ func buildEnvConfigData(set workloads.ReplicatedStateMachine) map[string]string
 	uid := string(set.UID)
 	strReplicas := strconv.Itoa(int(*set.Spec.Replicas))
 	generateReplicaEnv := func(prefix string) {
-		for i := 0; i < int(*set.Spec.Replicas); i++ {
+		// avoid to build too many envs
+		maxEnv := 64
+		for i := 0; i < int(*set.Spec.Replicas) && i < maxEnv; i++ {
```

Comment:
So some have ENV while else not ? What if the script or image  reference the non-existing ENVs ?

Reply:
The problem of `KB_<pod ordinal>_Hostname` ENVs is, too many ENVs will be generated if the `Replicas` is large. The ConfigMap object holding the ENVs will exceed the object limit of K8s, and the Containers can't run well too.

So I think this is a design fault of `KB_<pod ordinal>_Hostname` and give a quick hack fix here.